### ISSUE TITLE
Full customization of tables

### ISSUE THREAD

**Issue by rkiss** [lensapp/lens#2237]
**What would you like to be added**:
I would like to be able to customize each table that is displayed in Lens (Nodes, Pods, Events, Jobs ...):
- What columns are visible
- Additional columns at each table. For example for pods I would like to see the node name or resource limits and requests
- Able to filter for each column like you can do in Excel: I would like to see only the events from a particular node or of particular type. A regexp filter will do.
- Ability to resize and reorder columns
- Sorting (already exists)

**Why is this needed**:
The ability to customize the look and feel of Lens will greatly improve usability.

**Environment you are Lens application on:**

- Kubernetes distribution: [EKS, AKS, GKE, Rancher, ...] 
- Desktop OS: Linux

**Comment by jim-docker** [comment]
Some investigative work is underway (https://github.com/lensapp/lens/pull/1823, looking at react-table)

**Comment by pashevskii** [comment]
> Some investigative work is underway (#1823, looking at react-table)

This PR looks dead, #2146 is more hopeful :)

**Comment by guanzo** [comment]
yes please. I want to see the Job duration. Like what you get from `kubectl get job`

**Comment by maartengo** [comment]
>Able to filter for each column like you can do in Excel: I would like to see only the events from a particular node or of particular type.

This would be very helpful. Something like `name:aks-usernode*` when filtering on node names. 

I would also like a way to filter on nodes while in the pods workloads page, either through a drop-down or filter like `node:aks-usernode*`

_('aks-usernode' is an example node name)_

**Comment by ghost** [comment]
Please give a priority for this feature!
